src/work/SimpleDAO.sol

```diff
@@ -6,47 +6,67 @@ pragma solidity ^0.8.0;
  * @dev A more extensive version of a DAO that allows members to create proposals, vote on them, and execute proposals with various functionalities.
  */
 contract SimpleDAO {
+    enum ProposalType { Regular, Membership, Funding }
+    enum ProposalState { Pending, Active, Succeeded, Defeated, Executed }
+
     struct Proposal {
+        ProposalType proposalType;
+        ProposalState proposalState;
         string description;
-        bool executed;
         uint yesVotes;
         uint noVotes;
         mapping(address => bool) voted;
         uint creationTime;
         uint expiryTime;
         address target;
         bytes callData;
+        uint deposit;
+    }
+
+    struct Member {
+        bool isMember;
+        uint votingPower;
+        address delegate;
     }
 
     address public owner;
-    mapping(address => bool) public members;
-    mapping(address => uint) public votingPower;
-    uint public totalMembers; // Track total members for quorum calculation
+    mapping(address => Member) public members;
```

Comment:
This doesnt have to be public, maps dont get returned to be human readable. We can have a public method instead to return whether an address passed in is a member or not